test/arrayops.jl

```diff
@@ -2059,12 +2059,12 @@ struct F21666{T <: Base.TypeArithmetic}
     x::Float32
 end
 
+Base.TypeArithmetic(::Type{F21666{T}}) where {T} = T()
```

Comment:
Would it make sense to leave a note for the posterity why `struct F21666` and these definitions should be outside of `@testset`?